release/models/mirror/openconfig-mirror.yang

```diff
@@ -0,0 +1,241 @@
+module openconfig-mirror {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/mirror";
+
+  prefix "oc-mirror";
+
+  // import some basic types
+  import openconfig-extensions {
+    prefix oc-ext;
+  }
+  import openconfig-interfaces {
+    prefix oc-if;
+  }
+  import openconfig-acl {
+    prefix oc-acl;
+  }
+  import openconfig-inet-types {
+    prefix oc-inet;
+  }
+  import openconfig-network-instance {
+    prefix oc-ni;
+  }
+
+  // meta
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+     www.openconfig.net";
+
+  description
+    "Model for configuration and operational state data related
+     to capturing packets and sending them to some destination
+     for debugging or monitoring.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2024-06-12" {
+    description
+      "Initial revision";
+    reference
+      "0.1.0";
+  }
+
+  // extension statements
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // feature statements
+  // identity statements
+  // typedef statements
+  typedef mirror-action {
+    type enumeration {
```

Comment:
I'll add a port option.

Regarding `mirror-action` I am ok to change this typdef name.  I didn't use "destination" because that overlapped with a destination IP address (if using the remote option).  But maybe that is not too confusing, since `destination-address` is used for the IP addr.   But maybe it's ok:

* `/mirroring/sessions/session/destination` - PORT, FILE, REMOTE
* `/mirroring/sessions/session/destination-address` - the destination IP address for REMOTE

Any other suggestions for renaming mirror-action?

Reply:
Maybe `mirror-redirection`?